### DISCUSSION TITLE
How does CNPG manage PostgreSQL major upgrades on CNPG-managed clusters?

### DISCUSSION THREAD

**Discussion by ch680351034** [cloudnative-pg/cloudnative-pg#3329] [Q&A] [answered]
We recently migrated the postgres-13 databases from zalando operator to cnpg operator by using pg_basebackup method.

Now we want to test the upgrade from pg-13 (cnpg cluster) to pg-15.

Any documentation to follow to do the upgrade along with extensions

**Comment by rouke-broersma** [comment]
If you read https://cloudnative-pg.io/documentation/current/database_import/ they mention major upgrades as a scenario for using database import. Seems to me like you should simply create a new cluster for the new version and import your old database into that to perform the upgrade.

**Reply by gbartolini** [reply, marked as answer]
The current state is explained here: https://www.enterprisedb.com/blog/current-state-major-postgresql-upgrades-cloudnativepg-kubernetes. Thanks.

**Reply by gbartolini** [reply]
This is a more recent update: https://www.gabrielebartolini.it/articles/2024/03/cloudnativepg-recipe-5-how-to-migrate-your-postgresql-database-in-kubernetes-with-~0-downtime-from-anywhere/
